### PR TITLE
Update XLNet Model (einsum, outputs' format, k_head_r shape)

### PR DESCRIPTION
**Pull request by yingyibiao**
### PR types

Bug fixes, Function optimization
### PR changes

Models APIs Docs
### Description

1.  replace einsum ops
2. unify output format with other models
3. fix a bug when reshaping k_head_r's shape

### CODE REVIEW

File: paddlenlp/transformers/xlnet/modeling.py [hunk 1]
```diff
@@ -244,7 +216,7 @@ def forward(
             k_head_r = paddle.matmul(r, self.r)
             k_head_r = paddle.reshape(
                 k_head_r,
-                shape=[cat.shape[0], cat.shape[1], self.n_head, self.d_head])
+                shape=[r.shape[0], r.shape[1], self.n_head, self.d_head])
```
Comment (guoshengCS) [inline paddlenlp/transformers/xlnet/modeling.py:219]: 下面的q_head_h是否也应类似的将`cat.shape[0], cat.shape[1]`改为`h.shape[0], h.shape[1]`

File: paddlenlp/transformers/xlnet/modeling.py [hunk 2]
```diff
@@ -1318,16 +1271,12 @@ def forward(
...
-            output_hidden_states (bool, optional):
-                See :class:`XLNetModel`.
             return_dict (bool, optional):
                 See :class:`XLNetModel`.
 
         Returns:
-            tuple or dict: A tuple with items: (`output`, `new_mems`, `hidden_states`, `attentions`)
-            or a dict with key-value pairs: {"last_hidden_state": `output`, "mems": `new_mems`,
+            Tensor or dict: Output tensor `output` or a dict with key-value pairs:
+            {"last_hidden_state": `output`, "mems": `new_mems`,
```
Comment (guoshengCS) [inline paddlenlp/transformers/xlnet/modeling.py]: 这里的返回的dict的key中包含logits吧，last_hidden_state应该是logits吧

File: paddlenlp/transformers/xlnet/modeling.py [hunk 3]
```diff
@@ -1450,16 +1396,12 @@ def forward(
...
-            output_hidden_states (bool, optional):
-                See :class:`XLNetModel`.
             return_dict (bool, optional):
                 See :class:`XLNetModel`.
 
         Returns:
-            tuple or dict: A tuple with items: (`output`, `new_mems`, `hidden_states`, `attentions`)
-            or a dict with key-value pairs: {"last_hidden_state": `output`, "mems": `new_mems`,
+            Tensor or dict: Output tensor `output` or a dict with key-value pairs:
+             {"last_hidden_state": `output`, "mems": `new_mems`,
```
Comment (guoshengCS) [inline paddlenlp/transformers/xlnet/modeling.py]: 同上，这里的返回的dict的key中包含logits吧，last_hidden_state应该是logits吧

File: paddlenlp/transformers/xlnet/modeling.py (hunk 1 above)
Comment (yingyibiao) [reply paddlenlp/transformers/xlnet/modeling.py:219]: done

File: paddlenlp/transformers/xlnet/modeling.py (hunk 2 above)
Comment (yingyibiao) [reply paddlenlp/transformers/xlnet/modeling.py]: done

File: paddlenlp/transformers/xlnet/modeling.py (hunk 3 above)
Comment (yingyibiao) [reply paddlenlp/transformers/xlnet/modeling.py]: done

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**